src/github/pullRequestModel.ts

```diff
@@ -1566,42 +1567,56 @@ export class PullRequestModel extends IssueModel<PullRequest> implements IPullRe
 		}
 	}
 
-	async markFileAsViewed(filePathOrSubpath: string): Promise<void> {
-		const { mutate, schema } = await this.githubRepository.ensure();
-		const fileName = filePathOrSubpath.startsWith(this.githubRepository.rootUri.path) ?
-			filePathOrSubpath.substring(this.githubRepository.rootUri.path.length + 1) : filePathOrSubpath;
-		await mutate<void>({
-			mutation: schema.MarkFileAsViewed,
-			variables: {
-				input: {
-					path: fileName,
-					pullRequestId: this.graphNodeId,
-				},
-			},
-		});
+	async markFilesAsViewed(filePathOrSubpaths: string[]): Promise<void> {
+		const { mutate } = await this.githubRepository.ensure();
+		const pullRequestId = this.graphNodeId;
+
+		const filenames = filePathOrSubpaths
+			.map((f) =>
+				f.startsWith(this.githubRepository.rootUri.path)
+					? f.substring(this.githubRepository.rootUri.path.length + 1)
+					: f
+			);
+
+		const mutation = gql`mutation BatchMarkFileAsViewedInline {
+			${filenames.map((path) =>
+				`${path.split('/').join('')}: markFileAsViewed(
+					input: {path: "${path}", pullRequestId: "${pullRequestId}"}
+				) { clientMutationId }
+				`
+			)}
+		}`;
```

Comment:
Wasn't too sure about inlining these, but given that they are dynamic it seemed like the way to go

Reply:
I'm not a fan of this inlining. Is it possible to do without the inlining and instead have a new query in `queries.gql`?